internal/blockchain/ethereum/ethereum.go

```diff
@@ -49,21 +50,22 @@ const (
 )
 
 type Ethereum struct {
-	ctx              context.Context
-	topic            string
-	fireflyContract  string
-	fireflyFromBlock string
-	fireflyMux       sync.Mutex
-	prefixShort      string
-	prefixLong       string
-	capabilities     *blockchain.Capabilities
-	callbacks        blockchain.Callbacks
-	client           *resty.Client
-	fftmClient       *resty.Client
-	streams          *streamManager
-	initInfo         struct {
-		stream *eventStream
-		sub    *subscription
+	ctx             context.Context
+	topic           string
+	prefixShort     string
+	prefixLong      string
+	capabilities    *blockchain.Capabilities
+	callbacks       blockchain.Callbacks
+	client          *resty.Client
+	fftmClient      *resty.Client
+	streams         *streamManager
+	streamID        string
+	fireflyContract struct {
+		mux            sync.Mutex
+		address        string
+		fromBlock      string
+		networkVersion int
```

Comment:
Note comment above over `int` vs. semver. Probably an open discussion, rather than a proposal.

Reply:
As I ramble with half-formed thoughts here... I wonder whether it's actually a network-options block, where we've reserved a byte for a version integer, but other contract implementations could express additional details in that.

Reply:
I guess I may be completely overthinking this, as fundamentally this is just a constructor argument with a getter on the contract. So any extensibility to other options might be contract specific.

I'm just feeling an extensibility worry, and wondering if there's some way to do something now that would give future-us some more flexibility for some (currently unknown) migration challenge ahead, that could avoid the need for another contract migration by allowing more options to be encoded in the existing contract without change.

Reply:
I think these are all good points, and probably worthy of a lengthy discussion in their own right. Perhaps if an `int` is "good enough" for this PR, that discussion can be deferred to a follow-up issue or FIR? We can make a separate call on whether it's all part of 1.1 (and thus can break/replace all the logic here).

Reply:
I've opened #846 as a placeholder to continue this discussion.